src/client.js

```diff
@@ -42,15 +42,43 @@ export default class Client {
     }, options);
 
     if (options.apiHostname && isString(options.apiHostname)) {
-      const apiHostnameParsed = url.parse(options.apiHostname);
-      options.apiProtocol = apiHostnameParsed.protocol || 'https:';
-      options.apiHost = apiHostnameParsed.host;
+      let apiHostname = options.apiHostname;
+
+      if (/^https?:\/\//i.test(apiHostname) === false) {
```

Comment:
If someone uses `http://` the `if` statement will equal true. The regex `/^https?:\/\//i` checks for `http://` and `https://` strings. 

However, I am open to take it a step further and parsing the url before to determine the protocol being used.